onnxruntime/core/providers/ngraph/ngraph_custom_op.cc

```diff
@@ -141,11 +172,10 @@ void NGRAPHCustomOp::Initialize(const OrtCustomOpApi* api, OrtKernelContext* con
 Status NGRAPHCustomOp::Compute(const OrtCustomOpApi* api, OrtKernelContext* context) const {
   Ort::CustomOpApi ort{*api};
 
-  //TODO: Minimize locked region
-  std::lock_guard<std::mutex> lock(compute_lock_);
-
   // Initialize nGraph function if it is not already initialized.
+  compute_lock_.lock();
```

Comment:
using std::lock_guard is cleaner and also ensures the lock is released when exception is hit. 
prefer that to manual lock/unlock (especially because of all the unlocks in the catch blocks)

btw, do you have a sense of how much the previous implementation, where the lock was held throughout Compute() was impacting performance?

Reply:
Certainly in agreement that the manual lock is not the cleanest, but doing so does seem to reduce latency by ~15% for some models. Other models don't show as much benefit from it. Do you think the performance is worth the messy code?

Reply:
Changed in the latest commit.